src/Text/Pandoc/App/CommandLineOptions.hs

```diff
@@ -727,10 +728,14 @@ options =
 
      , Option "" ["csl"]
                  (ReqArg
-                  (\arg opt ->
-                     return opt{ optMetadata =
-                                   addMeta "csl" (normalizePath arg) $
-                                   optMetadata opt })
+                  (\arg opt -> do
+                     let (replaced, meta) = replaceMeta "csl" (normalizePath arg) $
+                                   optMetadata opt
+                     when (isJust replaced) $
+                      runIOorExplode $ report $ DuplicateMeta $
+                          "Only one CSL file can be specified."
+                            <> maybe "" (" Ignoring " <>) (metaValueToText $ fromJust replaced)
+                     return opt{ optMetadata = meta})
```

Comment:
This use of isJust and fromJust is better avoided (since fromJust is a partial function).
Better:
```
case replaced of
  Nothing -> ...
  Just f -> ...
```